components/onboarding/onboardingContext.js

```diff
@@ -0,0 +1,204 @@
+import { createContext, useContext, useEffect, useState } from 'react';
+import { useRouter } from 'next/router';
+import { driver } from 'driver.js'; // import driver.js
+
+import 'driver.js/dist/driver.css'; //import css
+import onboardingScriptGeneric from './onboardingScriptGeneric.json';
+import onboardingScriptCustom from './onboardingScriptCustom.json';
+
+const OnboardingContext = createContext(undefined);
+
+function useOnboardingProvider() {
+    const router = useRouter();
+    const [active, setActive] = useState(false);
+
+    const [currentRoute, setCurrentRoute] = useState('');
+    const [nextRouteName, setNextRouteName] = useState('');
+
+    const [prevRouteName, setPrevRouteName] = useState('');
+
+    const [currentRouteIndex, setCurrentRouteIndex] = useState(0);
+
+    const [currentSteps, setCurrentSteps] = useState([]);
+
+    const [currentStep, setCurrentStep] = useState(0);
+    const [hasPrev, setHasPrev] = useState(false);
+    const [hasNext, setHasNext] = useState(false);
+    const [currentStepDescription, setCurrentStepDescription] = useState('');
+    const [currentStepTitle, setCurrentStepTitle] = useState('');
+
+    const [driverJsConfig, setDriverJsConfig] = useState(null);
+
+    const [tourInstance, setTourInstance] = useState(null);
+
+    const onboardingScript = onboardingScriptCustom?.length > 0 ? onboardingScriptCustom : onboardingScriptGeneric;
+
+    const isScriptCustom = onboardingScriptCustom?.length > 0;
```

Comment:
We're just exporting this const to know which translation file to use? in that case I would say we can just use the conditional from line 34 instead.